File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/_tokens/alpha/AlphaRoles.kt
@file:Suppress("ktlint:standard:package-name")

package com.perrystreet.woof.designsystem.atomic._tokens.alpha

import androidx.compose.runtime.Immutable
import com.perrystreet.woof.designsystem.atomic._primitives.colors.AlphaPrimitives

@Immutable
data class AlphaRoles(
    val enabled: Float,
    val disabled: Float,
    val pressed: Float,
    val heroDim: Float,
    val shimmerLow: Float,
    val shimmerLowMedium: Float,
) {
    companion object {
        val Default = AlphaRoles(
            enabled = AlphaPrimitives.Max,
            disabled = AlphaPrimitives.Medium,
            pressed = AlphaPrimitives.Medium,
            heroDim = AlphaPrimitives.High,
            shimmerLow = AlphaPrimitives.Low,
            shimmerLowMedium = AlphaPrimitives.LowMedium,
        )
    }
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/_Tokens/Alpha/AlphaRoles.swift
import Foundation

public struct AlphaRoles {
    public let enabled: Double
    public let disabled: Double
    public let pressed: Double
    public let heroDim: Double
    public let shimmerLow: Double
    public let shimmerLowMedium: Double

    public static let `default` = AlphaRoles(
        enabled: AlphaPrimitives.max,
        disabled: AlphaPrimitives.medium,
        pressed: AlphaPrimitives.medium,
        heroDim: AlphaPrimitives.high,
        shimmerLow: AlphaPrimitives.low,
        shimmerLowMedium: AlphaPrimitives.lowMedium
    )
}
